roles/ks-core/prepare/files/ks-init/role-templates.yaml

```
@@ -3175,7 +3185,7 @@ role:
   metadata:
     annotations:
       iam.kubesphere.io/dependencies: '["role-template-view-pipelines", "role-template-view-pipelineruns"]'
-      iam.kubesphere.io/module: PipelineRuns Management
+      iam.kubesphere.io/module: Pipelines Management
```

/cc @harrisonliu5 Pipelines Management is different from  PipelineRuns Management, right? I don't think it should be modified here.

---

IMO, PipelineRuns depend on Pipelines, and cannot be dissociated from Pipelines. So I put all PipelineRuns' roles into `Pipelines Management` category.

---

It's ok for me at least now. PipelineRun cannot exist without Pipeline.

---

Make sense, role templates `role-template-manage-pipelineruns`  and `role-template-view-pipelineruns`  should be removed.

---

I agree with @wansir.

Hi @LinuxSuRen , what do you reckon about removing the two role templates?

---

I didn't see a reason why we should remove `role-template-manage-pipelineruns` and `role-template-view-pipelineruns`.

The permission of a PipelineRun should be able to control. For example, user A might only be able to trigger or view a set of pipelines.

---

The scenario which someone might be able to view Pipelines and trigger or approve the Pipeline runs make sense to me.

---

> s

Isn't `role-template-manage-pipelineruns` same as `role-template-manage-pipelines`? It's duplicated.

---

They are different.

---

Please see:

- https://github.com/kubesphere/ks-devops/blob/3d202827fb4f74d60a9a7277cc60af1ea9a093c0/pkg/api/devops/v1alpha3/pipelinerun_types.go
- https://github.com/kubesphere/ks-devops/blob/3d202827fb4f74d60a9a7277cc60af1ea9a093c0/pkg/api/devops/v1alpha3/pipeline_types.go

---

`role-template-manage-pipelines` determines who can trigger the pipeline.

---

At present, `role-template-manage-pipelines` module depends on `role-template-manage-pipelineruns`.

---

> `role-template-manage-pipelines` determines who can trigger the pipeline.

This is not correct. The CRD `Pipeline` can only represent a Pipeline. The CRD `PipelineRun` represents how to trigger a Pipeline.

---

```suggestion
      iam.kubesphere.io/module: PipelineRuns Management
```

Got it, do not modify the module name in this line. The role template is `PipelineRuns Management`, ks-console depends on the fixed module name.